### ISSUE TITLE
fix: history report in CI includes too many branches/commits

### ISSUE THREAD

**Issue by BenjaSanchez** [opencobra/memote#498]
#### Problem description

We have recently started the implementation of memote as CI for [yeast-GEM](https://github.com/SysBioChalmers/yeast-GEM/pull/162), a repo currently consisting of ~400 commits. However, ~5000 commits were included in the report, as it can be seen in the [travis log](https://travis-ci.org/SysBioChalmers/yeast-GEM/builds/430619010#L1314), making visualization [quite hard](http://sysbiochalmers.github.io/yeast-GEM/history_report.html). Many of these commits/branches could easily be skipped and not shown at all when using memote as CI, including:

* **Tags**: makes sense to highlight single commits that have a tag; however currently a full branch with all history for each tag is created. As we already have 15 tagged releases, that is a lot of extra commits.
* **Local branches**: if memote is being called from Travis, then the repo will have no local branches. However, the branch from which the build was made is being considered as a "local" branch, showing up duplicated in the history report (`feat/memote` & `origin/feat/memote`).
* **HEAD branch**: `origin/HEAD` will always be redundant with another branch (in the case of yeast-GEM, `origin/master`), so I also think it makes sense to skip it in the case of CI builds.
* **Feature branches**: is it necessary to show the whole past of each feature branch? Why not since the moment they were forked? This would also save a lot of repetition.
* **Github pages**: Finally, a branch for `origin/gh-pages` is also displayed, but I think this should be skipped as well as it is only used for generating the reports.

**Comment by Midnighter** [comment]
Thank you for the many great points of feedback.

* **tags**: I agree it makes sense to show a singular data point only.
* **local branches**: As discussed, an option to only consider remote branches makes sense.
* **HEAD**: Should be excluded, no question.
* **feature branches**: Question is, does the repetition hurt since they are overlaying data points? For example, if you only show two branches, wouldn't you want to see their entire history?
* **GitHub pages**: I was already excluding the local deployment branch but forgot to exclude the remote deployment branch :smiley: Thanks for catching that.

**Comment by BenjaSanchez** [comment]
@Midnighter yeah maybe the repetition in feature branches is not too bad (we try to not keep to many anyways), I was just giving ideas on how to reduce repetition. But let's see first how much we save with the other changes, maybe it's already enough :)
